File: frontend/src/types/audit-log.ts
export interface AuditLog {
  id: number;
  company_id: number | null;
  user_id: number | null;
  action: string;
  table_name: string;
  record_id: number | null;
  details: Record<string, any> | null;
  timestamp: string;
}

File: backend/app/schemas/audit_log.py
from datetime import datetime
from typing import Optional, Dict, Any
from pydantic import BaseModel


class AuditLogResponse(BaseModel):
    id: int
    company_id: Optional[int] = None
    user_id: Optional[int] = None
    action: str
    table_name: str
    record_id: Optional[int] = None
    details: Optional[Dict[str, Any]] = None
    timestamp: datetime

    model_config = {"from_attributes": True}
